lib/std/priority_dequeue.zig

```diff
@@ -8,27 +8,29 @@ const expectEqual = testing.expectEqual;
 const expectError = testing.expectError;
 
 /// Priority Dequeue for storing generic data. Initialize with `init`.
-/// Provide `compareFn` that returns `Order.lt` when its first
-/// argument should get min-popped before its second argument,
+/// Provide `compareFn` that returns `Order.lt` when its second
+/// argument should get min-popped before its third argument,
 /// `Order.eq` if the arguments are of equal priority, or `Order.gt`
-/// if the second argument should be min-popped first.
+/// if the third argument should be min-popped second.
 /// Popping the max element works in reverse. For example,
 /// to make `popMin` return the smallest number, provide
-/// `fn lessThan(a: T, b: T) Order { return std.math.order(a, b); }`
-pub fn PriorityDequeue(comptime T: type, comptime compareFn: fn (T, T) Order) type {
+/// `fn lessThan(context: void, a: T, b: T) Order { _ = context; return std.math.order(a, b); }`
+pub fn PriorityDequeue(comptime T: type, comptime Context: type, comptime compareFn: fn (context: Context, a: T, b: T) Order) type {
```

Comment:
Should `Context` be named something that specifically associates it with the comparator? `CompareFnContext` ? `SortContext` ?